pkg/rtcp/packet.go

```diff
@@ -5,3 +5,81 @@ type Packet interface {
 	Marshal() ([]byte, error)
 	Unmarshal(rawPacket []byte) error
 }
+
+// PacketWithHeader is a pair to represent an RTCP header and it's
+// packet's polymorphic parsed and unparsed forms.
+type PacketWithHeader struct {
+	Header
```

Comment:
I've sort of done what I described above -- the Unmarshal factory returns the header and the packet (because you need to generate both at the same time)